audit/conn/domainsocket.go

```diff
@@ -0,0 +1,77 @@
+//go:build linux || darwin
+
+package conn
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/microsoft/go-otel-audit/audit/conn/internal/writer"
+)
+
+// Compile check on interface implementation.
+var _ Audit = DomainSocketConn{}
+
+// DomainSocketConn represents a connection to a remote audit server via a unix domain socket
+// set to SOCK_STREAM. This implements conn.Audit interface.
+type DomainSocketConn struct {
+	*writer.Conn
```

Comment:
Learning question for me: The TCP Connection you've got is buffered by you're not buffering this. Is that because you know the latency here is low because it's a DomainSocket?

Reply:
I see what's happening now from the above comment.  Your confusing the "writer.go" with "tcp.go", my guess is you walked away and came back to the review :)  

Done this a time or two.

TCP and Domainsocket are exactly the same except construction options that create the net.Conn object.

The Writer is buffered and it buffers for both types.  Domain sockets are a ton faster and I probably could have eschewed using buffering there due to the speed difference. But having a common place for buffering is useful and the other problem is that even if you are using DS, you are really talking TCP in the end.  That's because what is reading DS is simply forwarding over TCP.  So in essence, it's a TCP connection and will need the same buffering scheme.

Reply:
Doh this is what I get for finishing a review up at the end of a day. This all makes sense, sorry about the confusion!